src/sentry/web/frontend/auth_organization_login.py

```diff
@@ -44,11 +44,14 @@ def handle_basic_auth(self, request, organization):
             user = register_form.save()
             user.send_confirm_emails(is_new_user=True)
 
-            defaults = {
-                'role': 'member',
-            }
+            try:
+                OrganizationMember.objects.get(email=user.email)
```

Comment:
This query is most likely going to be wrong. I think we just want to check if this user is already a member of the organization they are joining at a minimum. In theory, you just want:

```python
if not OrganizationMember.objects.filter(user=user, organization=organization).exists():
    ....
```

But I'll need to confirm exactly.

Reply:
@mattrobenolt do we need an organization check? Shouldn't we map the user if he is part of any organizations?

Reply:
I'll have to look more closely to see exactly what's going on here. I'm not too intimately familiar, it just seems wrong at a glance. :)